packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -94,9 +95,15 @@ library LibProposing {
 
             if (blobHash == 0) revert L1_NO_BLOB_FOUND();
         } else {
+            // The proposer must be an Externally Owned Account (EOA) for
+            // calldata usage. This ensures that the transaction is not an
+            // internal one, making calldata retrieval more straightforward.
+            if (msg.sender != tx.origin) revert L1_PROPOSER_NOT_EOA();
```

Comment:
I guess we want to eliminate this:
`EOA` ->` Contract A` -> `Contract B` (TaikoL1)

Wondering what happens if the `Contract A `would use `delegatecall()` to preserve the EOA context with rubbish (non findeable, or not easily decodeable) TX list.

```

taikoL1.delegatecall(
            abi.encodeWithSignature("proposeBlock(bytes, bytes)", params, made_up_txn_list)
        );
```

Reply:
`TransparentProxyUpgradeable` is also using `delegateCall()`, and since `TaikoL1` is upgradeable, the proxy also uses the delegateCall() pattern. 
So others could easily do that too, no ?

Reply:
@dantaik @davidtaikocha 
Maybe something like this in TaikoL1.sol ? 🤔  (But if this is a proxy, then it may complicate things, hmm)
https://ethereum.stackexchange.com/questions/128924/is-there-a-way-to-check-if-a-call-is-a-call-or-a-delegatecall

Reply:
> I guess we want to eliminate this: `EOA` ->` Contract A` -> `Contract B` (TaikoL1)
> 
> Wondering what happens if the `Contract A` would use `delegatecall()` to preserve the EOA context with rubbish (non findeable, or not easily decodeable) TX list.
> 
> ```
> 
> taikoL1.delegatecall(
>             abi.encodeWithSignature("proposeBlock(bytes, bytes)", params, made_up_txn_list)
>         );
> ```

using delegateCall in contractA will make the sender to be the contractA address and the origin to be the user, so the if-statement will return true to revert.

Reply:
> https://ethereum.stackexchange.com/questions/128924/is-there-a-way-to-check-if-a-call-is-a-call-or-a-delegatecall

this approach prevents our contract to be called as code by third party contracts. I'm not sure we need to do it...

Reply:
We should use this feature to prevent our code to be used if our code has a restrictive license, so other people cannot deploy a copy of our code but want to call our deployed contract using delegate call.

Reply:
> using delegateCall in contractA will make the sender to be the contractA address 

delegateCall() will preserve the context of msg.sender, msg.data, mag.value, so in this case EOA would be msg.sender 🤔  ?

Reply:
If you are interacting with contractA which delegatecall TaikoL1, then your transaction is not a Taiko transaction (although the code is the same), our client won't listen to contractA's events...

Reply:
> If you are interacting with contractA which delegatecall TaikoL1, then your transaction is not a Taiko transaction (although the code is the same), our client won't listen to contractA's events...

Yeah, right ! Thankfully it seems false alarm then. ;)

Reply:
Ah more discussion about this here! I want to raise another concern (https://github.com/taikoxyz/taiko-mono/pull/15090#discussion_r1377821887):
> This may break integrating Taiko with for example PBS because that will likely require some external contract, unless we do something like allowing an extra callback to be specified in the propose function that can do some extra logic.

Reply:
I responded there and here is a PR for checking the EOA differently https://github.com/taikoxyz/taiko-mono/pull/15095